fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -2689,4 +2693,28 @@ public void addFinishedQueryDetail() {
         QueryDetailQueue.addQueryDetail(queryDetail);
     }
 
+    public int getExecTimeout() {
+        if (parsedStmt instanceof DmlStmt) {
```

Comment:
Is it better to provide a abstract getTimeout method in StatementBase, with some sub-classes implementing it?

Reply:
done